pkgs/development/tools/gauge/plugins/xml-report/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib
+, makeGaugePlugin
+}:
+makeGaugePlugin {
+  name = "xml-report";
+  data = lib.importJSON ./data.json;
+
+  repo = "getgauge/xml-report";
+  releasePrefix = "xml-report-";
+  systemMap = {
```

Comment:
It seems like this map is very similar for every plugin except that not all support all platforms. Maybe the presence of an a download url could be enough to map this automatically?

Reply:
done!